vm_automation/esxiVm.py

```diff
@@ -222,29 +222,25 @@ def __init__(self, serverObject, vmObject):
 
     def checkTools(self, waitForTools = True):
         """
-        THERE HAS TO BE A BETTER WAY TO DO THIS....
-        WHEN I WROTE IT, THERE WERE ONLY 2 POSSIBLE VALUES, ON AND NOT ON YET...
-        THERE ARE A FEW OPTIONS I FOUND LATER THAT ARE THE EQUIVALENT OF "NEVER COMING ON, YOU BETTER QUIT"
-            DIFFICULT TO EXIT FROM WITHIN A FUNCTION, AND RETURNING False MEANS "WAIT"
-            PROBABLY NEED TO TRUNCATE THIS TO
-                CHECKING FOR STATE,
-                ENCAPSULATE WITH try/except
-                RETURN STRING OR FLAG FOR "READY" "NOT READY" "FAULT"
+        I WISH THIS COULD BE BINARY, BUT IT NEEDS THREE VALUES...
+        TOOLS_NOT_READY:     VMWARE_TOOLS IS NOT READY, BUT MAY BECOME READY (THE CALLER SHOULD KEEP TRYING)
+        TOOLS_READY:         VMWARE_TOOLS IS READY
+        TOOLS_NOT_INSTALLED: VMWARE TOOLS IS NOT READY AND NEVER WILL BE
         """
         tools_status = self.vmObject.guest.toolsStatus
         if tools_status == 'toolsNotRunning':
-            retVal = False
+            retVal = 'TOOLS_NOT_READY'

```

Comment:
I would really prefer constants to hardcoded strings.  I know this is a scripting language but anything that can make IDEs better at helping generate "correct" code is more easily maintainable.